src/box/txn.c

```diff
@@ -734,6 +734,8 @@ txn_complete_fail(struct txn *txn)
 	if (txn->limbo_entry != NULL) {
 		assert(txn_has_flag(txn, TXN_WAIT_SYNC));
 		txn_limbo_abort(&txn_limbo, txn->limbo_entry);
+		txn_limbo_update_confirmed_entry_after_rollback(
+			&txn_limbo, txn->limbo_entry->lsn);
```

Comment:
Lets update the confirmed entry right inside of `txn_limbo_abort()`. It doesn't look right that the limbo's state gets broken after the abort and needs the explicit manual "repairing" via this new additional call.

Reply:
Fixed.